app/services/verify_link_service.rb

```diff
@@ -30,10 +30,11 @@ def link_back_present?
 
     if links.any? { |link| link['href'].downcase == @link_back.downcase }
       true
+    elsif link.any? { |link| link_redirects_back?(link['href']) }
+      true
     elsif links.empty?
       false
     else
-      link_redirects_back?(links.first['href'])
```

Comment:
Ah. I think this was a fix for Twitter putting t.co links on their profiles back when they still added `rel=me` on their page. However, your change basically sends a request to every link on the page! That's a lot of requests and a lot of time.

Reply:
Wait, no - this would only send a request to every `rel="me"` link on the page, as is defined in line 29:
https://github.com/mastodon/mastodon/blob/b303eacb9be49ed9ba9300b4c6b38be22f91aaec/app/services/verify_link_service.rb#L29

If you *did* have a lot of `rel="me"` links on your page, this still definitely would take up a lot of requests and time (so perhaps stop after 10 links?).

Reply:
Could  link_redirects_back first check to see if test_url is already equal to @link_back before trying the request to resolve any link shorteners? Is there any downside to not following the link?

Reply:
Links are already tested for exact match in the `links.any? { |link| link['href'].downcase == @link_back.downcase }` line of `link_back_present?`, so `link_redirects_back?` is only called when there is no direct match.

Reply:
That being said, this checks for `https://instance.tld/@:username`, while `https://instance.tld/users/:username` is also a valid URL for that user. Maybe both should be checked.